src/Actions.js

```diff
@@ -3,20 +3,28 @@ import { createEmail } from './EmailUtil'
 
 const API_ENDPOINT = process.env.REACT_APP_ENDPOINT
 
-function joinGroupMe(gmToken, callback){
+async function joinGroupMe(gmToken, callback){
   if (!gapi.auth2.getAuthInstance().isSignedIn.get()) {
       
     gapi.auth2.getAuthInstance().signIn().then(() => {joinGroupMe(gmToken, callback)})
     return
   }
+
+  var blacklisted = await isBlacklisted(gmToken, "groupme ID")
+  console.log(blacklisted);
+  if (blacklisted) {
+    alert("You are not authorized to join the GroupMe due to past behavior");
+    return
+  }
```

Comment:
Instead of creating a separate endpoint for "isBlacklisted", let's return that error message in the 'addtogroupme' endpoint if the user is blacklisted

Reply:
then the function would also no longer have to be async